Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py

```diff
@@ -1716,34 +1717,54 @@ def fetch_last_emails(
     demisto.debug(f"Finished getting the folder named {folder_name} by path")
     log_memory()
     if since_datetime:
-        if incident_filter == MODIFIED_FILTER:
-            qs = qs.filter(last_modified_time__gte=since_datetime)
-        else:  # default to "received" time
+        if incident_filter == RECEIVED_FILTER:
             qs = qs.filter(datetime_received__gte=since_datetime)
+        else:
+            qs = qs.filter(last_modified_time__gte=since_datetime)
     else:
         tz = EWSTimeZone("UTC")
         first_fetch_datetime = dateparser.parse(FETCH_TIME)
         assert first_fetch_datetime is not None
         first_fetch_ews_datetime = EWSDateTime.from_datetime(first_fetch_datetime.replace(tzinfo=tz))
-        qs = qs.filter(last_modified_time__gte=first_fetch_ews_datetime)
         demisto.debug(f"{first_fetch_ews_datetime=}")
+        if incident_filter == RECEIVED_FILTER:
+            qs = qs.filter(datetime_received__gte=first_fetch_ews_datetime)
+        else:
+            qs = qs.filter(last_modified_time__gte=first_fetch_ews_datetime)
+
     qs = qs.filter().only(*[x.name for x in Message.FIELDS if x.name.lower() != "mime_content"])
-    qs = qs.filter().order_by("datetime_received")
+    if incident_filter == RECEIVED_FILTER:
+        qs = qs.filter().order_by("datetime_received")
+    else:
+        qs = qs.filter().order_by("last_modified_time")
+
     result = []
-    exclude_ids = exclude_ids if exclude_ids else set()
+    exclude_ids = exclude_ids if exclude_ids else dict()
     demisto.debug(f"{APP_NAME} - Exclude ID list: {exclude_ids}")
     qs.chunk_size = min(client.max_fetch, 100)
     qs.page_size = min(client.max_fetch, 100)
     demisto.debug("Before iterating on queryset")
     demisto.debug(f"Size of the queryset object in fetch-incidents: {sys.getsizeof(qs)}")
     for item in qs:
         demisto.debug("next iteration of the queryset in fetch-incidents")
-        if isinstance(item, Message) and item.message_id not in exclude_ids:
+        if isinstance(item, Message):
+            if item.message_id in exclude_ids:
+                demisto.debug(
+                    f"prev: {exclude_ids.get(item.message_id)}, current: {item.datetime_created.ewsformat() if incident_filter == RECEIVED_FILTER else item.last_modified_time.ewsformat()}")
+                if not exclude_ids.get(item.message_id) or exclude_ids.get(item.message_id) >= (
+                item.datetime_created.ewsformat() if incident_filter == RECEIVED_FILTER else item.last_modified_time.ewsformat()):
```

Comment:
lets simplify this code, and store the relevant time field result in a specific var, and use it from this point on...